src/libraries/Microsoft.Extensions.Hosting.Systemd/src/SystemdHelpers.cs

```diff
@@ -30,6 +33,15 @@ private static bool CheckParentIsSystemd()
                 return false;
             }
 
+            // To support containerized systemd services, check if we're the main process (PID 1)
+            // and if there are systemd environment variables defined for notifying the service
+            // manager, or passing listen handles.
+            if (Environment.ProcessId == 1)
+            {
+                return !string.IsNullOrEmpty(Environment.GetEnvironmentVariable(NOTIFY_SOCKET)) ||
+                    !string.IsNullOrEmpty(Environment.GetEnvironmentVariable(LISTEN_PID));
+            }
+
```

Comment:
(nit) The name of this method is `CheckParentIsSystemd()`, but this is doing more than that. Can we update this method name to be something more like `GetIsSystemdService()`?

Reply:
Good spot, next commit will fix.